Added new kb article dataform-net-maui-validation (#1076)
telerik/maui-docs: knowledge-base/dataform-net-maui-validation.md

page_title: How to Add Validation to DataForm in .NET MAUI Using Data Annotations
slug: dataform-net-maui-validation
tags: dataform, .net maui, validation, data annotations, regex
res_type: kb
---

## Environment

| Versions | Product | Author | 
| --- | --- | ---- | 
| 9.0.0 | Telerik UI for .NET MAUI DataForm | [Dobrinka Yordanova](https://www.telerik.com/blogs/author/dobrinka-yordanova) | 

## Description

When working with DataForm in .NET MAUI, there might be requirements to validate the input data, such as checking if an entry is a valid email address or phone number. The question arises on how to implement these validations without manually coding standard inputs for each field type.

This knowledge base article also answers the following questions:
- How to validate phone numbers in DataForm?
- How to validate email addresses in DataForm?
- How to use regular expressions for validation in DataForm?

## Solution

To enable validation in the DataForm for .NET MAUI, utilize the data annotations provided by the `System.ComponentModel.DataAnnotations` namespace. These annotations can trigger validation on various events such as lost focus, property changed, or manually. 

> Here are examples on how to apply different validation rules:

Reviewer: Don't introduce document sections with an introductory sentence, as it will only be visible for the first section that comes after it.
```suggestion
The examples in this article show how to apply different validation rules.
```
